File: 算法训练营/03-数组链表跳表/026/双指针.go
/*
 * @lc app=leetcode.cn id=26 lang=golang
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
func removeDuplicates(nums []int) int {
	length := len(nums)
	if length < 2 {
		return length
	}
	
	cur, seek := 0, 1
	for ; seek < length; seek++ {
		if nums[cur] == nums[seek] {
			continue
		}
		// 需要交换
		if cur + 1 != seek {
			nums[cur+1] = nums[seek]
		}
		cur++
	}
	return cur+1
}
// @lc code=end

File: 算法训练营/03-数组链表跳表/283/双指针.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if len(nums) <= 1:
            return 
        
        prev, next = 0, 1
        while next < len(nums):
            if nums[prev] == 0 and nums[next] == 0:
                next += 1
                continue

            if nums[prev] == 0 and nums[next] != 0:
                nums[prev], nums[next] = nums[next], nums[prev]
            
            prev += 1
            next += 1
# @lc code=end